File: Lab2/PartB/Point.cpp
#include "Point.h"
#include <iostream>
#include <math.h>

using namespace std;

Point::Point()
{
    xM = 0;
    yM = 0;
    counter++;
    idM = 1000 + counter;
}

Point::Point(double x, double y)
{
    xM = x;
    yM = y;
    counter++;
    idM = 1000 + counter;
}

Point::Point(const Point &source)
{
    xM = source.xM;
    yM = source.yM;
    counter++;
    idM = 1000 + counter;
}

Point &Point::operator=(const Point &rhs)
{
    if (this != &rhs)
    {
        xM = rhs.xM;
        yM = rhs.yM;
    }
    return *this;
}

Point::~Point()
{
    counter--;
}

double Point::x() const
{
    return xM;
}

double Point::y() const
{
    return yM;
}

void Point::setx(double x)
{
    xM = x;
}

void Point::sety(double y)
{
    yM = y;
}

void Point::display() const
{
    cout << "X-coordinate:" << xM << endl
         << "Y-coordinate:" << yM << endl;
}

double Point::distance(const Point &p) const
{
    double xDiff = xM - p.xM;
    double yDiff = yM - p.yM;
    return sqrt(xDiff * xDiff + yDiff * yDiff);
}

double Point::distance(const Point &a, const Point &b)
{
    double xDiff = a.xM - b.xM;
    double yDiff = a.yM - b.yM;
    return sqrt(xDiff * xDiff + yDiff * yDiff);
}

int Point::count()
{
    return counter;
}

int Point::counter = 0;

int Point::id() const
{
    return idM;
}

File: Lab4/ExD/EXD/shapes/Point.java
package shapes;

public class Point{
	private double x_coordinate, y_coordinate;
	static int counter = 0;
	String id;
	
	public Point(double a, double b)
	{
		x_coordinate = a;
		y_coordinate = b;
		id = "P" + ++counter;
	}
	
	public String toString()   
	{
		String s;
		s = "\nPoint Id: " + id + "\nX_coordinate: " + x_coordinate +  "\nY-coordinate: " + y_coordinate;
		return s;
	}
	
	public double  getx()   
	{
		return x_coordinate;
	}
	
	void  setx(double newvalue)
	{
		x_coordinate = newvalue;
	}
	
	public double  gety()   
	{
		return y_coordinate;
	}
	
	public void  sety(double newvalue)
	{
		y_coordinate = newvalue;
	}
	
	public double  distance(Point  other)
	{
		double dist_x = other.x_coordinate - x_coordinate;
		double dist_y = other.y_coordinate - y_coordinate;
		
		return (Math.sqrt(Math.pow(dist_x, 2) + Math.pow(dist_y, 2)));
	}
	
	static double  distance (Point  that, Point  other)
	{
		double dist_x = other.x_coordinate - that.x_coordinate;
		double dist_y = other.y_coordinate - that.y_coordinate;
		
		return (Math.sqrt(Math.pow(dist_x, 2) + Math.pow(dist_y, 2)));
	}
	
	public static int count()
	{
		return counter;
	}
	
	public Object clone() throws CloneNotSupportedException {
		return super.clone();
		
	}
	
	public static void main(String [] args)
	{
		Point a = new Point (5, 6);
		Point b = new Point (45, 69);
		System.out.println(a.distance(b));
		Point.distance(a, b);
		System.out.println(a);
		
	}
}